src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -129,49 +130,239 @@ public static void Construction_MultiDimensionalArray()
             Assert.Equal(array.GetValue(0, 0, 0, 2), 3);
         }
 
-        public static IEnumerable<object[]> BinarySearch_NonGeneric_TestData()
+        public static IEnumerable<object[]> BinarySearch_TestData()
         {
             string[] stringArray = new string[] { null, "aa", "bb", "bb", "cc", "dd", "ee" };
+            string stringDummy = "";
```

Comment:
Yeah - its for generic inference. I actually made a PR to xunit yesterday to fix this: https://github.com/xunit/xunit/pull/965